sh/scenarios/regression_4771.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/utils/main.sh
+source "$NCTL"/sh/views/utils.sh
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+source "$NCTL"/sh/scenarios/common/itst.sh
+
+#######################################
+# This test checks if the entry point of an installed contact can successfully be invoked.
+#
+# It covers against the regression captured in the #4771 ticket.
+#######################################
+function main() {
+    log "------------------------------------------------------------"
+    log "Regression 4771 test begins"
+    log "------------------------------------------------------------"
+
+    do_await_genesis_era_to_complete
+    TX_HASH=$(install_contract)
+    log "Install contract transaction hash: $TX_HASH"
+    await_n_blocks 2
+    ENTITY_CONTRACT=$(get_entity_contract $TX_HASH)
+    log "Contract entity: $ENTITY_CONTRACT"
+    INVOKE_ENTRY_POINT_RESULT=$(call_entry_point $ENTITY_CONTRACT)
+    log "Result of entry point invocation: $INVOKE_ENTRY_POINT_RESULT"
+
+    if [[ "$INVOKE_ENTRY_POINT_RESULT" == *"no such contract at hash"* ]]; then
+        log "Test failed: contract not found"
+        exit 1
+    fi
+
+    log "------------------------------------------------------------"
+    log "Regression 4771"
+    log "------------------------------------------------------------"
+
+    exit 0
+}
+
+function install_contract() {
+    local CHAIN_NAME
+    local GAS_PAYMENT
+    local NODE_ADDRESS
+    local PATH_TO_CLIENT
+    local PATH_TO_CONTRACT
+    local CONTRACT_OWNER_SECRET_KEY
+    local CONTRACT_ARG_MESSAGE="Hello Dolly"
+
+    # Set standard deploy parameters.
+    CHAIN_NAME=$(get_chain_name)
+    GAS_PAYMENT=10000000000000
+    NODE_ADDRESS=$(get_node_address_rpc)
+    PATH_TO_CLIENT=$(get_path_to_client)
+
+    # Set contract path.
+    PATH_TO_CONTRACT="./smart_contracts/gh_4771_regression.wasm"
+    if [ ! -f "$PATH_TO_CONTRACT" ]; then
+        echo "ERROR: The gh_4771_regression.wasm binary file cannot be found.  Please compile it and move it to the following directory: ./sh/scenarios/smart_contracts/"
+        return
+    fi
+
+    # Set contract owner secret key.
+    CONTRACT_OWNER_SECRET_KEY=$(get_path_to_secret_key "$NCTL_ACCOUNT_TYPE_FAUCET")
+
+    # Dispatch deploy (hits node api).
+    TRANSACTION_INSTALL_RESULT=$(
+        $PATH_TO_CLIENT put-transaction session \
+            --chain-name "$CHAIN_NAME" \
+            --node-address "$NODE_ADDRESS" \
+            --transaction-path "$PATH_TO_CONTRACT" \
+            --session-entry-point call \
+            --session-arg "name:string='TEST_GH4771'" \
+            --category install-upgrade \
+            --payment-amount 150000000000 \
```

Comment:
nit: not sure if the client will error out if the `--payment-amount` is not specified here but this parameter is not used when pricing mode is set to fixed.

Reply:
I'd rather keep this as is for now, because the client is still under development and is a subject to change. Once the client for 2.0 is stabilized, we can revisit that, wdyt?

Reply:
yup, we can change it in a latter commit